lib/puppet_x/certs/common.rb

```diff
@@ -43,25 +43,37 @@ module Common
         newparam(:password_file)
 
         newparam(:ca) do
+          isrequired
+
           validate do |value|
             ca_resource = resource.catalog.resource(value.to_s)
-            if ca_resource && ca_resource.class.to_s != 'Puppet::Type::Ca'
-              raise ArgumentError, "Expected Ca resource, got #{ca_resource.class} #{ca_resource.inspect}"
+            if ca_resource
+              # rspec-puppet presents Puppet::Resource instances
```

Comment:
What does this comment mean? Is this like "we have to do the next line because of our test infrastructure" ?

Reply:
Pretty much yes.